capitals.py

```diff
@@ -1,15 +1,23 @@
-def letters_count():
-    print('Please write your text')
-    input_text = input()
-    count_lower = 0
+def count_letters(string):
     count_upper = 0
+    count_lower = 0
 
-    for i in input_text:
+    for i in string:
         if (i.islower()):
             count_lower = count_lower + 1 
         elif (i.isupper()):
             count_upper = count_upper + 1
-    print('There are {} uppercase letters'.format(count_upper))
-    print('There are {} lowercase letters'.format.(count_lower))
 
-letters_count()
+    return (count_upper, count_lower)
+
+def main():
```

Comment:
deci in main se cheama count_letters - foarte logic! :D :D I like this structure more
main e un fel de aranjare frumoasa a core-ului programului, un fel de cluster pentru toate functiile din acest fisier?
si la sfarsit se cheama doar main ca asa e mai comod/frumos?

Reply:
main e punctul de intrare în program, de unde el începe execuția. În mod obișnuit python execută toate instrucțiunile care nu-s în funcții la rând. Dar în cazul dat noi nu avem instrucțiuni care nu-s în funcții, doar `if __name__ == "__main__":
    main()`.  Și ceea ce face bucata asta doar verifică dacă fișierul a fost chemat pentru interpretare și nu pentru import.

Adica în mod normal când ai instrucțiuni înafara funcșiilor și faci `$python script.py` sau `import script` în alt fișier rezultatul va fi același, ele vor fi executate. Dar dacă vrei la import să nu fie executate atunci trebuie să înțelegi cum e apelat fișierul și bucata cu if __name__ anume asta face, dacă e interpretat și nu importat el va executa main, dacă nu ap nu.